hyperapp.d.ts

```diff
@@ -0,0 +1,68 @@
+export as namespace hyperapp
+
+/** @namespace [Virtual DOM] */
+
+/** The Virtual DOM representation of an Element
+ *
+ * @memberOf [Virtual DOM]
+*/
+export interface VirtualNode<Data> {
+  tag: string
+  data: Data
+  children: VirtualNodeChild<{} | null>[]
+}
+
+/** In the Virtual DOM a Child could be either a Virtual Node or a string
+ *
+ * @memberOf [Virtual DOM]
+*/
+export type VirtualNodeChild<Data> = VirtualNode<Data> | string
+
+/** A Component is a function that return a custom Virtual Node
```

Comment:
I'd put this comment inside the interface, on top of the method definition, and add param documentation. It provides better intellisense in VSCode